src/internal/m365/collection/exchange/container_resolver_test.go

```diff
@@ -53,6 +55,316 @@ func strPtr(s string) *string {
+				mcg,
+				"userID",
+				test.includes,
+				test.excludes)
+			require.NoError(t, err, clues.ToCore(err))
+
+			item := rcr.ItemByID(test.itemID)
+			if test.expectFound {
+				require.NotNil(t, item)
+				assert.Equal(t, test.itemID, ptr.Val(item.GetId()), "returned item ID")
+			} else {
+				assert.Nil(t, item, "unexpected item returned: %+v", item)
+			}
+		})
+	}
+}
+
+func (suite *RankedContainerResolverUnitSuite) TestItems() {
+	// Containers available to operate on directly in tests.
+	const (
+		id1         = "id1"
+		id2         = "id2"
+		id3         = "id3"
+		idErr       = "idErr"
+		idEmpty     = "idEmpty"
+		idNotInBase = "idNotInBase"
+	)
+
+	mcg := mockContainerGetter{
+		itemsByID: map[string]containerGetterRes{
+			id1:         {c: mockContainer{id: ptr.To(id1)}},
+			id2:         {c: mockContainer{id: ptr.To(id2)}},
+			id3:         {c: mockContainer{id: ptr.To(id3)}},
+			idErr:       {err: assert.AnError},
+			idEmpty:     {c: mockContainer{}},
+			idNotInBase: {c: mockContainer{id: ptr.To(idNotInBase)}},
+		},
+	}
+
+	// Configure base containers we want.
+	mcr := &mockContainerResolver{
+		containersByID: map[string]graph.CachedContainer{
+			id1: &mockCachedContainer{id: id1},
+			id2: &mockCachedContainer{id: id2},
+			id3: &mockCachedContainer{id: id3},
+		},
+	}
+
+	// Add a bunch more containers so we're more likely to get a random order from
+	// the map.
+	var otherContainerIDs []string
+
+	for i := 0; i < 100; i++ {
+		id := fmt.Sprintf("extra-id%d", i)
+		mcr.containersByID[id] = &mockCachedContainer{id: id}
+		otherContainerIDs = append(otherContainerIDs, id)
+	}
+
+	table := []struct {
+		name     string
+		includes []string
+		excludes []string
+		// expectPrefix is the prefix of container IDs that should be in the result.
+		expectPrefix []string
+		// expectExtraUnordered allows specifying additional IDs in the set of IDs
+		// available to work with that will appear in the unordered set. For
+		// example, if only id1 was part of includes and excludes was empty then id2
+		// and id3 should appear somewhere in the output but don't have a particluar
+		// order requirement.
+		expectExtraUnordered []string
+		expectErr            assert.ErrorAssertionFunc
+	}{
+		{
+			name:         "ReturnsRankedItems",
+			includes:     []string{id2, id1, id3},
+			expectPrefix: []string{id2, id1, id3},
+			expectErr:    assert.NoError,
+		},
+		{
+			name:                 "ReturnsRankedItems SomeUnordered",
+			includes:             []string{id2, id1},
+			expectPrefix:         []string{id2, id1},
+			expectExtraUnordered: []string{id3},
+			expectErr:            assert.NoError,
+		},
+		{
+			name:         "ReturnsRankedItems SomeExcluded",
+			includes:     []string{id2},
+			excludes:     []string{id3, id1},
+			expectPrefix: []string{id2},
+			expectErr:    assert.NoError,
+		},
+		{
+			name:                 "ReturnsRankedItems SomeIncludesNotInBase",
+			includes:             []string{id2, id1, idNotInBase},
```

Comment:
Curious, is it realistically possible to have `idNotInBase` in `includes` but not in base?